Formula/b/bornagain.rb

```diff
@@ -0,0 +1,173 @@
+  end
+
+  def make_shims(venv_py)
+    # produce helper scripts
+
+    brew_bin_pth = (HOMEBREW_PREFIX/"bin/").realpath
+
+    # extract the installation location of the BornAgain package via `pip show`
+    # eg. 'Location: /opt/homebrew/Cellar/bornagain/22.0/venv/lib/python3.12/site-packages'
+    location_rx = /^\s*Location\s*:\s*(.+)\s*/i
+    info = `#{venv_py} -m pip show bornagain`
+    ba_loc = info.match(location_rx)[1]
+
+    shim = prefix/"shim"
+    mkdir_p shim
+
+    # script to run BornAgain GUI with embedded Python
+    gui_script = %Q(\
+     #! /bin/sh
+
+     # run BornAgain with Python support
+     export PYTHONPATH="#{ba_loc}:${PYTHONPATH}"
+
+     #{bin}/bornagain
+    )
+
+    ba_gui_cmd = shim/"bornagain_gui"
+    File.write(ba_gui_cmd, gui_script, mode: "w")
+    # make the script executable for all users
+    chmod 0755, ba_gui_cmd
+
+    # script to use BornAgain Python package
+    pyenv_script = %Q(\
+     #! /bin/sh
+
+     # add BornAgain package path to PYTHONPATH
+     export PYTHONPATH="#{ba_loc}:${PYTHONPATH}"
+
+     # usage:
+     # $ source #{brew_bin_pth}/bornagain_py"
+    )
+
+    ba_pyenv_cmd = shim/"bornagain_py"
+    f = File.write(ba_pyenv_cmd, pyenv_script, mode: "w")
+    chmod 0755, ba_pyenv_cmd
+
+    # script to display info about BornAgain
+    ba_share = share/"BornAgain"
+
+    info_script = %Q(\
+     #! /bin/sh
+
+     echo '-------------- BornAgain ---------------'
+     echo '* BornAgain: simulate and fit reflection and scattering'
+     echo '* Homepage: <http://www.bornagainproject.org>'
+     echo '* Copyright Forschungszentrum Jülich GmbH 2018'
+     echo '----------------------------------------'
+     echo '* Version #{version}: #{prefix}/'
+     echo '* Python wheel: #{ba_share}/wheel/'
+     echo '* Python examples: #{ba_share}/Examples/'
+     echo '* Build log: #{ba_share}/bornagain_build.log'
+     echo '----------------------------------------'
+     echo '* Run GUI with Python support:'
+     echo '$ bornagain'
+     echo '----------------------------------------'
+     echo "* Use BornAgain as a Python package:"
+     echo '$ source #{brew_bin_pth}/bornagain_py'
+     echo '----------------------------------------'
+    )
+
+    ba_info_cmd = shim/"bornagain_info"
+    File.write(ba_info_cmd, info_script, mode: "w")
+    chmod 0755, ba_info_cmd
+  end
+
+  def install
+    ff_cmake_dir = Formula["libformfactor"].prefix/"cmake/"
+    heinz_cmake_dir = Formula["libheinz"].prefix/"cmake/"
+
+    # Build a Python virtual environment with the required packages
+    venv = virtualenv_create(prefix/"venv", "python3")
+    venv_root = venv.instance_variable_get(:@venv_root)
+    venv_py = venv_root/"bin/python3"
+    venv.pip_install resources
+
+    system "cmake", "-S", ".", "-B", "build",
+           *std_cmake_args, "-DBA_TESTS=OFF",
+           "-DBORNAGAIN_PYTHON=ON",
+           "-DBA_PY_PACK=ON",
+           "-DCMAKE_PREFIX_PATH=#{ff_cmake_dir};#{heinz_cmake_dir};",
```

Comment:
This shouldn't be needed
```suggestion
```

Reply:
On my Mac Mini machine (with `Homebrew 4.4.29-9-gc370200`, `CMake 3.31.6`), CMake cannot find `libformfactor` and `libheinz` without the hint `-DCMAKE_PREFIX_PATH=#{ff_cmake_dir};#{heinz_cmake_dir};`. Therefore, they should be kept, otherwise, I receive the following error:

```
Last 15 lines from /Users/admin/Library/Logs/Homebrew/bornagain/03.cmake:
  Could not find a package configuration file provided by "LibHeinz"
  (requested version 2.0.0) with any of the following names:

    LibHeinzConfig.cmake
    libheinz-config.cmake

  Add the installation prefix of "LibHeinz" to CMAKE_PREFIX_PATH or set
  "LibHeinz_DIR" to a directory containing one of the above files.  If
  "LibHeinz" provides a separate development package or SDK, be sure it has
  been installed.

Last 15 lines from /Users/admin/Library/Logs/Homebrew/bornagain/03.cmake:
  Could not find a package configuration file provided by "formfactor"
  (requested version 0.3.0) with any of the following names:

    formfactorConfig.cmake
    formfactor-config.cmake

  Add the installation prefix of "formfactor" to CMAKE_PREFIX_PATH or set
  "formfactor_DIR" to a directory containing one of the above files.  If
  "formfactor" provides a separate development package or SDK, be sure it has
  been installed.
```

Reply:
In that case you can use:
```suggestion
           "-DLibHeinz_DIR=#{Formula["libheinz"].opt_prefix}/cmake/",
           "-Dformfactor_DIR=#{Formula["formfactor"].opt_prefix}/cmake/",
```

Reply:
Done. See revision 3e2f2b72734.